internal/signer/sort_test.go

```diff
@@ -0,0 +1,71 @@
+package signer_test
+
+import (
+	"encoding/json"
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/theupdateframework/go-tuf/data"
+	"github.com/theupdateframework/go-tuf/internal/signer"
+	"github.com/theupdateframework/go-tuf/pkg/keys"
+)
+
+type mockSigner struct {
+	value json.RawMessage
+}
+
+func (s *mockSigner) MarshalPrivateKey() (*data.PrivateKey, error) {
+	panic("not implemented")
+	return nil, nil
+}
+
+func (s *mockSigner) UnmarshalPrivateKey(key *data.PrivateKey) error {
+	panic("not implemented")
+	return nil
+}
+
+func (s *mockSigner) PublicData() *data.PublicKey {
+	return &data.PublicKey{
+		Type:       "mock",
+		Scheme:     "mock",
+		Algorithms: []string{"mock"},
+		Value:      s.value,
+	}
+}
+func (s *mockSigner) SignMessage(message []byte) ([]byte, error) {
+	panic("not implemented")
+	return nil, nil
+}
+
+func TestSignerSortByIDs(t *testing.T) {
+	s1 := &mockSigner{
+		value: json.RawMessage(`{"mock": 1}`),
+	}
+	s2 := &mockSigner{
+		value: json.RawMessage(`{"mock": 2}`),
+	}
+	s3 := &mockSigner{
+		value: json.RawMessage(`{"mock": 3}`),
+	}
+	s4 := &mockSigner{
+		value: json.RawMessage(`{"mock": 4}`),
+	}
+	s5 := &mockSigner{
+		value: json.RawMessage(`{"mock": 5}`),
+	}
+
+	s := []keys.Signer{
+		s1, s2, s3, s4, s5,
+	}
+
+	sort.Sort(signer.ByIDs(s))
+
+	sorted := []keys.Signer{
```

Comment:
Sorry I did not realize this part last time. I think the reason for this random order is because the IDs is the hash of the value? This makes this test very unreadable. 
Can you please mock the IDs in the PublicKey to return the exact value (instead of the hash of the value)? 
and in line 59 input a random order of the signers ... in line 65 change the variable to expected and orders to be s1, s2, s3, s4, s5?

Reply:
Hmmm I think that can not quite address the issue.